go/obsclient/clientutil/retry_strategy.go

```diff
@@ -0,0 +1,50 @@
+package clientutil
+
+import (
+	"fmt"
+	"time"
+)
+
+// retryStrategy interface allows for flexible strategies for retrying/polling functions
+type retryStrategy interface {
```

Comment:
Maybe this is a bit over-engineered? I didn't want callers to have to pass in too many parameters for the `AwaitReceipt` (it needs context, client, hash) so then I wanted to be able to have a default for the timeout, the wait interval etc. without compromising on flexibility if they wanted to specify...

This would allow for backing off strategies etc. but maybe it's too much at this stage? Open to ideas.